src/HonzaBotner.Discord.Services/Commands/PollCommands.cs

```diff
@@ -146,21 +149,23 @@ private async Task PollHelpAsync(CommandContext ctx)
             return;
         }
 
-        DiscordRole modRole = (await ctx.Client.GetGuildAsync(ctx.Guild.Id)).GetRole(_options.ModRoleId);
-        AbcPoll poll = new(originalMessage);
-
-        if (poll.AuthorMention != ctx.Member.Mention && !ctx.Member.Roles.Contains(modRole))
-        {
-            await ctx.RespondAsync("You are not authorized to edit this poll");
-            return;
-        }
-
         try
         {
-            await new AbcPoll(originalMessage).AddOptionsAsync(ctx.Client, options.ToList());
+            DiscordRole modRole = (await _guildProvider.GetCurrentGuildAsync()).GetRole(_options.ModRoleId);
+
+            // I am sorry, due to DSharpPlus' caching logic, this mess is necessary
+            AbcPoll poll = new (await (await ctx.Client.GetChannelAsync(ctx.Channel.Id)).GetMessageAsync(ctx.Message.ReferencedMessage.Id));
```

Comment:
Lol

Reply:
🤮

Reply:
Have to use these two methods, they are used only there, didn't want to crate special variables for them... Do you want me to do something with it, or is it commitable?